src/lib/multipart/MultipartHttpClient.ts

```diff
@@ -1,22 +1,56 @@
-import { HttpMethod } from 'simple-http-request-builder';
-import { HttpPromise, unwrapHttpPromise } from '../promise/HttpPromise';
-import { networkErrorCatcher } from '../client/FetchClient';
-import {
-  genericError, HttpResponse, networkError, timeoutError,
-} from '../client/HttpResponse';
 import {
+  HttpMethod,
   MultipartHttpClient,
   MultipartHttpOptions,
   MultipartHttpRequest,
-} from './MultipartHttpRequest';
+} from 'simple-http-request-builder';
+import { Logger } from 'simple-logging-system';
+import {
+  FetchResponseHandler,
+  networkErrorCatcher,
+} from '../client/FetchClient';
+import {
+  genericError,
+  HttpResponse,
+  networkError,
+  timeoutError,
+  toErrorResponsePromise,
+} from '../client/HttpResponse';
+import { HttpPromise, unwrapHttpPromise } from '../promise/HttpPromise';
+
+const logger: Logger = new Logger('MultipartHttpClient');
+
+const createResponseFromXhr = (xhr: XMLHttpRequest): Response => {
+  // Extract headers from XMLHttpRequest
+  const headers = new Headers();
+  const rawHeaders = xhr.getAllResponseHeaders();
+  rawHeaders.trim().split(/[\r\n]+/).forEach((line) => {
```

Comment:
```suggestion
  const headers = new Headers();
  xhr.getAllResponseHeaders().trim().split(/[\r\n]+/).forEach((line) => {
```
- il manque le typage sur headers et line